pkg/networkservice/common/connect/server.go

```diff
@@ -155,26 +155,21 @@ func (c *connectServer) Close(ctx context.Context, conn *networkservice.Connecti
 		// In case of error, no pending state is propogated.
 		return nil, err
 	}
-	// Call next Close before calling client Close
```

Comment:
I've done this with purpose, since if remote nsmgr/endpoint will die, we need to cleanup on Close all our next chains in any case.

Reply:
OK... after a long discussion on slack, I think that we landed at:

Close is responsible for calling downstream Closes in the chain, even if it encounters an error in cleaning up its own state (in this case, client.Close() is cleaning up its own state).

To that end, I've updated this PR to call next.Server(ctx).Close() even if client.Close() returns an error.  @haiodo Does this handle your concerns?